File: Frontend/js/system.js
(async()=>{
 const s=await api.getSystemSettings();
 const range=document.getElementById("preference"), out=document.getElementById("preferenceValue");
 range.value=s.optimizationPreference; const update=()=>{const v=+range.value; out.textContent=v<50?`Cost-focused (${v}% emissions)`:v>50?`Emissions-focused (${100-v}% cost)`: "Balanced";}; update(); range.addEventListener("input",update);
 document.getElementById("reserveInput").value=s.minBatteryReserve;
 document.getElementById("dieselInput").value=s.dieselMaxOutput;
 document.getElementById("horizonInput").value=s.planningHorizon;
 document.getElementById("intervalInput").value=s.forecastUpdateInterval;
 document.getElementById("dieselAvailability").value=s.dieselAvailability;
 ["reserveInput","dieselInput","horizonInput","intervalInput","dieselAvailability"].forEach(id=>{
   document.getElementById(id).addEventListener("change",e=>{ document.getElementById(id+"State")?.replaceChildren(document.createTextNode(e.target.value)); });
 });
})()

File: api/schemas/system.py
from typing import Literal

from pydantic import BaseModel, Field


class SystemSettingsUpdate(BaseModel):
    optimization_preference: float = Field(ge=0, le=100)
    min_battery_reserve: float = Field(ge=0, le=90)
    diesel_max_output: float = Field(ge=0, le=1000)
    planning_horizon: int = Field(ge=6, le=24)
    diesel_availability: Literal["Available", "Unavailable"]
    forecast_update_interval: int = Field(ge=1, le=1440)
